crates/core/src/types.rs

```diff
@@ -68,89 +131,162 @@ pub struct CancelBundle {
+        skip_serializing_if = "Option::is_none"
+    )]
+    pub max_timestamp: Option<u64>,
 
-    pub fn uuid(&self) -> &Uuid {
-        &self.uuid
-    }
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub reverting_tx_hashes: Vec<TxHash>,
+
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub replacement_uuid: Option<String>,
+
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub dropping_tx_hashes: Vec<TxHash>,
+
+    pub meter_bundle_response: MeterBundleResponse,
+}
 
-    pub fn bundle_hash(&self) -> B256 {
+pub trait BundleTxs {
+    fn transactions(&self) -> &Vec<Recovered<OpTxEnvelope>>;
+}
+
+pub trait BundleExtensions {
+    fn bundle_hash(&self) -> B256;
+    fn txn_hashes(&self) -> Vec<TxHash>;
+    fn senders(&self) -> Vec<Address>;
+    fn gas_limit(&self) -> u64;
+    fn da_size(&self) -> u64;
+}
+
+impl<T: BundleTxs> BundleExtensions for T {
+    fn bundle_hash(&self) -> B256 {
+        let parsed = self.transactions();
         let mut concatenated = Vec::new();
-        for tx in self.transactions() {
+        for tx in parsed {
             concatenated.extend_from_slice(tx.tx_hash().as_slice());
         }
         keccak256(&concatenated)
     }
 
-    pub fn txn_hashes(&self) -> Vec<TxHash> {
+    fn txn_hashes(&self) -> Vec<TxHash> {
         self.transactions().iter().map(|t| t.tx_hash()).collect()
     }
 
-    pub fn bundle(&self) -> &Bundle {
-        &self.bundle
-    }
-
-    pub fn senders(&self) -> Vec<Address> {
+    fn senders(&self) -> Vec<Address> {
         self.transactions()
             .iter()
             .map(|t| t.recover_signer().unwrap())
             .collect()
     }
 
-    pub fn gas_limit(&self) -> u64 {
-        self.transactions.iter().map(|t| t.gas_limit()).sum()
+    fn gas_limit(&self) -> u64 {
+        self.transactions().iter().map(|t| t.gas_limit()).sum()
     }
 
-    pub fn da_size(&self) -> u64 {
-        self.transactions
+    fn da_size(&self) -> u64 {
+        self.transactions()
             .iter()
             .map(|t| tx_estimated_size_fjord_bytes(&t.encoded_2718()))
             .sum()
     }
 }
 
-#[derive(Debug, Clone, Serialize, Deserialize)]
+impl BundleTxs for ParsedBundle {
+    fn transactions(&self) -> &Vec<Recovered<OpTxEnvelope>> {
+        &self.txs
+    }
+}
+
+impl BundleTxs for AcceptedBundle {
+    fn transactions(&self) -> &Vec<Recovered<OpTxEnvelope>> {
+        &self.txs
+    }
+}
+
+impl AcceptedBundle {
+    pub fn load(
+        mut bundle: Bundle,
+        meter_bundle_response: MeterBundleResponse,
+    ) -> Result<Self, String> {
+        let uuid = bundle
+            .replacement_uuid
+            .clone()
+            .unwrap_or_else(|| Uuid::new_v4().to_string());
+
+        let uuid = Uuid::parse_str(uuid.as_str()).map_err(|_| format!("Invalid UUID: {uuid}"))?;
+
+        bundle.replacement_uuid = Some(uuid.to_string());
+
+        let txs: Vec<Recovered<OpTxEnvelope>> = bundle
```

Comment:
in a future PR, this logic is in both `.load()` and the `TryFrom`, we can probably type cast `Vec<Recovered<OpTxEnvelope>>` as `Transactions` and then implement a `TryFrom` to reduce duplicate logic